stl/inc/vector

```diff
@@ -1307,10 +1313,21 @@ public:
     template <class _Iter, enable_if_t<_Is_iterator_v<_Iter>, int> = 0>
     _CONSTEXPR20 void assign(_Iter _First, _Iter _Last) {
         _Adl_verify_range(_First, _Last);
+
+        if (_First == _Last) {
+            clear();
+            return;
+        }
+
         _Assign_range(_Get_unwrapped(_First), _Get_unwrapped(_Last), _Iter_cat_t<_Iter>{});

```

Comment:
Rather than changing the call sites, I would prefer to change `_Assign_range` to properly handle empty ranges.

Reply:
I thought it would be nice to avoid taking the iterator invalidation locks if at all possible.
Also the general case for `_Assign_range` is that it is passed a real range.

Reply:
I'm confused because `clear()` needs to take the lock to invalidate iterators.

Reply:
Empty ranges are perfectly cromulent ranges! :smile_cat:

I've pushed centralized changes to make `_ASAN_VECTOR_MODIFY(0)` a no-op (even when the vector has zero capacity and thus has null pointers), which avoids the need for extensive logic changes and will be robust during future refactoring.